src/ChatPage.cpp

```diff
@@ -1530,6 +1531,38 @@ ChatPage::initiateLogout()
         emit showOverlayProgressBar();
 }
 
+void
+ChatPage::query_keys(
```

Comment:
Well, as we discussed before we don't want to update the key cache every time we find a outdated user, we rather mark them outdated and then use this api rather than the old one, so it will update only when needed, i feel this is a better approach because in the outdated list some random usernames are popping so query keys for each one of them would decrease the performance.